src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,47 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private K[] keys;
+    private V[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_SIZE];
+        values = (V[]) new Object[MAX_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < MAX_SIZE; i++) {
+            if (key == keys[i] || key != null && key.equals(keys[i])) {
+                if (values[i] != null) {
+                    size--;
+                }
+                values[i] = value;
+                break;
+            }
```

Comment:
If we use return instead of break we don't have to decrement size that will be incremented further
```suggestion
            if (key == keys[i] || key != null && key.equals(keys[i])) {
                values[i] = value;
                return;
            }
```